website/en/docs/types/arrays.md

```diff
@@ -142,10 +142,11 @@ readonlyArray[0] = {x: 42}; // Error!
 readonlyArray[0].x = 42; // OK
 ```
 
-If you use `$ReadOnlyArray` instead of `Array` for arrays that are never mutated,
-you can prevent Flow from raising spurious errors. For mutable arrays of generic
-types that are set to arrays of specific types there's the possibility of adding
-values to it that don't match the type of the specifically-typed array.
+The main advantage to using `$ReadOnlyArray` instead of `Array` is that `$ReadOnlyArray`'s
+type parameter is *covariant* while `Array`'s type parameter is *invariant*. That means that
+`$ReadOnlyArray<number>` is a subtype of `$ReadOnlyArray<number | string>` while
+`Array<number>` is NOT a subtype of `Array<number | string>`. So it's often useful to use
+`$ReadOnlyArray` as a type annotation to allow various different types of arrays.
```

Comment:
> So it's often useful to use `$ReadOnlyArray` as a type annotation to allow various different types of arrays.

I find this sentence confusing, because “various different types of arrays” is ambiguous. I would word the sentence like this:

> So it's often useful to use `$ReadOnlyArray` in type annotations for arrays of various types of elements.

Reply:
Hmmm yeah, I could see that. I personally don't think it's that ambiguous given the context but the change would be strictly more clear.

This PR has been out for a while and I don't want to keep @reyronald going back and forth. Merging it as is but if you're so inclined @roryokane feel free to PR a change in the wording.

Reply:
@mrkev I made the PR with that change: https://github.com/facebook/flow/pull/6816